services/activities/resource_functions/schemas.go

```diff
@@ -0,0 +1,162 @@
+package resource_functions
+
+import (
+	"github.com/leanspace/terraform-provider-leanspace/helper"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+var validResourceFunctionTimeUnits = []string{"SECONDS", "MINUTES", "HOURS", "DAYS"}
+var validFormulaTypes = []string{"LINEAR"}
+
+var resourceFunctionSchema = map[string]*schema.Schema{
+	"id": {
+		Type:     schema.TypeString,
+		Computed: true,
+	},
+	"activity_definition_id": {
+		Type:         schema.TypeString,
+		Required:     true,
+		ForceNew:     true,
+		ValidateFunc: validation.IsUUID,
+	},
+	"resource_id": {
+		Type:         schema.TypeString,
+		Required:     true,
+		ForceNew:     true,
+		ValidateFunc: validation.IsUUID,
+	},
+	"name": {
+		Type:     schema.TypeString,
+		Optional: true,
+	},
+	"time_unit": {
+		Type:         schema.TypeString,
+		Required:     true,
+		ValidateFunc: validation.StringInSlice(validResourceFunctionTimeUnits, false),
+		Description:  helper.AllowedValuesToDescription(validResourceFunctionTimeUnits),
+	},
+	"formula": {
+		Type:     schema.TypeList,
+		Required: true,
+		MinItems: 1,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: formulaSchema,
+		},
+	},
+	"created_at": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "When it was created",
+	},
+	"created_by": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Who created it",
+	},
+	"last_modified_at": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "When it was last modified",
+	},
+	"last_modified_by": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Who modified it the last",
+	},
+}
+
+var formulaSchema = map[string]*schema.Schema{
+	"constant": {
+		Type:     schema.TypeFloat,
+		Required: true,
+	},
+	"rate": {
+		Type:     schema.TypeFloat,
+		Required: true,
+	},
+	"type": {
+		Type:         schema.TypeString,
+		Required:     true,
+		ValidateFunc: validation.StringInSlice(validFormulaTypes, false),
+		Description:  helper.AllowedValuesToDescription(validFormulaTypes),
+	},
+}
+
+var dataSourceFilterSchema = map[string]*schema.Schema{
+	"ids": {
```

Comment:
ids is already present in the default list of available filters so you don't have to put it here

Reply:
I feel like it's not the first you tell me this :-)